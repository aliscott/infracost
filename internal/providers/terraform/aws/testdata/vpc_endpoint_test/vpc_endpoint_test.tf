provider "aws" {
  region                      = "us-east-1" # <<<<< Try changing this to eu-west-1 to compare the costs
  skip_credentials_validation = true
  skip_requesting_account_id  = true
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
}

resource "aws_vpc_endpoint" "interface" {
  service_name = "com.amazonaws.region.ec2"
  vpc_id = "vpc-123456"
  vpc_endpoint_type = "Interface"
}

resource "aws_vpc_endpoint" "gateway_loadbalancer" {
  service_name = "com.amazonaws.region.ec2"
  vpc_id = "vpc-123456"
  vpc_endpoint_type = "GatewayLoadBalancer"
}

resource "aws_vpc_endpoint" "multiple_interfaces" {
  service_name = "com.amazonaws.region.ec2"
  vpc_id = "vpc-123456"
  vpc_endpoint_type = "Interface"
  subnet_ids = [
    "subnet-123456",
    "subnet-654321"
  ]
}
