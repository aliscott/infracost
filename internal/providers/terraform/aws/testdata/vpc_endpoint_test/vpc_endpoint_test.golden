
 Name                                   Monthly Qty  Unit         Monthly Cost 
                                                                               
 aws_vpc_endpoint.gateway_loadbalancer                                         
 ├─ Endpoint (GatewayLoadBalancer)              730  hours               $7.30 
 └─ Data processed                      Cost depends on usage: $0.0035 per GB  
                                                                               
 aws_vpc_endpoint.interface                                                    
 ├─ Endpoint (Interface)                        730  hours               $7.30 
 └─ Data processed                      Cost depends on usage: $0.01 per GB    
                                                                               
 aws_vpc_endpoint.multiple_interfaces                                          
 ├─ Endpoint (Interface)                      1,460  hours              $14.60 
 └─ Data processed                      Cost depends on usage: $0.01 per GB    
                                                                               
 PROJECT TOTAL                                                          $29.20 

----------------------------------
To estimate usage-based resources use --usage-file, see https://infracost.io/usage-file